Sources/Code/AnyMenuViewController.swift

```diff
@@ -309,14 +319,24 @@ public class AnyMenuViewController: UIViewController {
 
     /// Opens the menu.
     public func openMenu(completion: ((Bool) -> Void)? = nil) {
-        menuState = .open
```

Comment:
The delegate is called whenever the menu state was changed. This is because the menu state could also be changed by a pan gesture. In this case the menu state is changed by the animator after the gesture and any animations completing the  gesture are finished. So in my opinion setting the menu state before the animation even begins is the wrong behavior.
Though its also possible updating the menu state only in the animator. Also the menu state could be extended by a third (and a fourth) state which then reflects that the menu is in a transition state.

Reply:
Yes it is needed for the delegation

Reply:
It might be that the toggleMenu() function in the UIViewController extension is affected by setting the menu state in the completion handler